base/setup/lib/setuplib.c

```diff
@@ -712,6 +712,102 @@ InitSystemPartition(
     return TRUE;
 }
 
+
+#define IS_PATH_SEPARATOR(c)    ((c) == L'\\' || (c) == L'/')
+
+/**
+ * @brief
+ * Verify whether the given directory is suitable for ReactOS installation.
+ * Each path component must be a valid 8.3 name.
+ **/
+BOOLEAN
+IsValidInstallDirectory(
+    _In_ PCWSTR InstallDir)
+{
+    PCWCH p;
+
+    /* As with the NT installer, fail if directory is empty or "\\" */
+    p = InstallDir;
+    if (!*p || (IS_PATH_SEPARATOR(*p) && !*(p + 1)))
+        return FALSE;
+
+#if 0
+    /* Path must contain only valid characters (alpha-numeric, '.',
+     * '\\', '-' and '_'). Note that spaces are not accepted. */
+    for (p = InstallDir; *p; ++p)
+    {
+        if (!IS_VALID_INSTALL_PATH_CHAR(*p))
+            return FALSE;
+    }
+#endif
```

Comment:
Why is this in a `#if 0`?

Reply:
I disabled it because I wanted to exercise the other loop (the one that runs over the path sub-components) for the tests. I'll re-enable it when it's confirmed it does what's needed.

Reply:
Re-enabled now.